File: doctor/src/main/java/com/ygfh/doctor/ui/demo/TwoActivity.java
package com.ygfh.doctor.ui.demo;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.fh.baselib.http.BaseObserver;
import com.fh.baselib.utils.L;
import com.ygfh.doctor.R;
import com.ygfh.doctor.data.bean.Login;
import com.ygfh.doctor.data.bean.RxSchedulers;
import com.ygfh.doctor.net.DcServiceFactory;

import org.jetbrains.annotations.Nullable;

public class TwoActivity extends AppCompatActivity {
    private Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_two);
        context = this;
        initViews();
    }

    private void initViews() {
        findViewById(R.id.btn_login).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DcServiceFactory.INSTANCE.getService()
                        .login2()
                        .compose(RxSchedulers.ioMain(context))
                        .subscribe(new BaseObserver<Login>() {
                            @Override
                            public void onSuccess(@Nullable Login login) {
                                L.Companion.e("login:" + login);
                            }
                        });
            }
        });
    }
}

File: doctor/src/main/java/com/ygfh/doctor/ui/demo/TwoActivity.kt
package com.ygfh.doctor.ui.demo

import com.fh.baselib.mvp.MvpBaseActivity
import com.fh.baselib.utils.L
import com.ygfh.doctor.R
import kotlinx.android.synthetic.main.activity_two.*

class TwoActivity : MvpBaseActivity<TwoContract.TwoView, TwoPresenter>(), TwoContract.TwoView {

    override fun loginSuccess() {
        L.d("登录成功--------")

    }

    override fun loadStoreSuccess() {

    }

    override fun layoutId(): Int {
        return R.layout.activity_two
    }

    override fun initView() {
        btn_login.setOnClickListener {
            mPresenter = null
            //?.表示 不为空的时候执行
            mPresenter!!.login()
        }
    }

    override fun initListener() {

    }
}
